New serverless pattern - eventbridge-cloudwatch-cdk-java (#2211)
aws-samples/serverless-patterns: eventbridge-cloudwatch-cdk-java/README.md

# Amazon EventBridge to Amazon CloudWatch

> This project contains a sample AWS CDK template to create an EventBridge Rule, as well as, a CloudWatch Logs Group. The EventBridge Rule publishes matched events to CloudWatch Logs. In this example, the rule filters for specific attributes in the event before sending to the CloudWatch Logs target.

Reviewer: ```suggestion
This project contains a sample AWS CDK template to create an EventBridge Rule and a CloudWatch Logs log group. The EventBridge Rule publishes matched events to the log group. In this example, the rule filters for specific attributes in the event before sending to the target.
```

Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.

## Requirements

- [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
- [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
- [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
- [AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/cli.html) installed and configured
  
## Deployment Instructions

1. Clone the project to your local working directory
    ```
    git clone https://github.com/aws-samples/serverless-patterns
    ```

2. Change the working directory to this pattern's directory
    ```

>     cd serverless-patterns/eventbridge-cloudwatch-cdk-java

Reviewer: ```suggestion
    cd eventbridge-cloudwatch-cdk-java
```

Author: On this one if you where to clone the repo using the command in step 1 it with the default behaviour it would clone it into a directory `serverless-patterns`. 

I can add another step to cd into `serverless-patterns` if that makes it easier.

    ```

3. From the command line, configure AWS CDK:
   ```bash
   cdk bootstrap ACCOUNT-NUMBER/REGION # e.g.
   cdk bootstrap 1111111111/us-east-1
   cdk bootstrap --profile test 1111111111/us-east-1
   ```
4. From the command line, use AWS CDK to deploy the AWS resources for the pattern as specified in the `lib/cdk-stack.ts` file:
   ```bash
   cdk deploy
   ```

> 5. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.

Reviewer: ```suggestion
5. Note the outputs from the CDK deployment process. These contain the resource names and/or ARNs which are used for testing.
```

## How it works

The CDK stack deploys the resources and the IAM permissions required to run the application.

> The EventBridge rule filters the events based upon the defined criteria. When matching events are sent to EventBridge that trigger the rule, they are published to the CloudWatch Logs Group.

Reviewer: ```suggestion
The EventBridge rule filters the events based upon the defined criteria. Matched events are published to the CloudWatch Logs log group.
```

## Testing

Use the [AWS CLI](https://aws.amazon.com/cli/) to send a test event to EventBridge and observe the event delivered to the Amazon CloudWatch by reviewing the logs:

1. Send an event to EventBridge:

```sh
aws events put-events --entries file://event.json
```

> 2. Check the CloudWatch Logs, referencing the log group name from the Outputs section of the deployed Stack, to see an event matching this example:

Reviewer: ```suggestion
2. Check CloudWatch Logs, referencing the log group name from the stack's Outputs section, to see an event matching this example:
```
